File: javaSample/src/main/java/muxi/javasample/tasks/TransactionTask.java
package muxi.javasample.tasks;

import android.app.ProgressDialog;
import android.os.AsyncTask;

import muxi.payservices.sdk.service.IMPSManager;
import muxi.javasample.AppConstants;
import muxi.payservices.sdk.data.MPSTransaction;


public class TransactionTask extends AsyncTask<Void,Void,Void> {


    private ProgressDialog progressDialog;
    private IMPSManager mpsManager;
    private MPSTransaction mpsTransaction;
    private AppConstants.TransactionState mTransactionState;


    public TransactionTask(ProgressDialog progressDialog,
                           IMPSManager mpsManager,
                           MPSTransaction mpsTransaction,
                           AppConstants.TransactionState mTransactionState
                           ){
        this.progressDialog = progressDialog;
        this.mpsManager = mpsManager;
        this.mpsTransaction = mpsTransaction;
        this.mTransactionState = mTransactionState;
    }

    @Override
    protected void onPreExecute() {
        super.onPreExecute();
        //Todo remover este marretado
        progressDialog.setMessage("Waiting pinpad...");
        progressDialog.setIndeterminate(false);
        progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
        progressDialog.setCancelable(false);
        progressDialog.show();

    }

    @Override
    protected Void doInBackground(Void... voids) {
        switch (mTransactionState){
            case payment:
                mpsManager.transaction(mpsTransaction);
                break;
            case cancel:
                mpsManager.cancelTransaction(mpsTransaction);
                break;
        }
        return null;
    }
}

File: kotlinSample/src/main/java/muxi/kotlinsample/ui/present_card/tasks/TransactionTask.kt
package muxi.kotlinsample.ui.present_card.tasks

import android.os.AsyncTask
import muxi.payservices.sdk.data.MPSTransaction
import muxi.kotlinsample.Constants
import muxi.payservices.sdk.service.IMPSManager

class TransactionTask(val mpsManager: IMPSManager,
                      val mpsTransaction: MPSTransaction,
                      val transactionState: Constants.TransactionState): AsyncTask<Void, Void, Void>() {

    override fun doInBackground(vararg params: Void?): Void? {
        when(transactionState){
            Constants.TransactionState.payment->
                mpsManager.transaction(mpsTransaction)
            Constants.TransactionState.cancel->
                mpsManager.cancelTransaction(mpsTransaction)
        }
        return null
    }
}